crypt.go

```diff
@@ -458,6 +458,38 @@ func DeactivateVolume(volumeName string) error {
 	return luks2Deactivate(volumeName)
 }
 
+// KDFOptions specifies parameters for the Argon2 KDF used by cryptsetup.
+type KDFOptions struct {
+	// MemoryKiB specifies the maximum memory cost in KiB when ForceIterations
+	// is zero. If ForceIterations is not zero, then this is used as the
+	// memory cost.
+	MemoryKiB int
+
+	// TargetDuration specifies the target duration for the KDF which
+	// is used to benchmark the time and memory cost parameters. If it
+	// is zero then the cryptsetup default is used. If ForceIterations
+	// is not zero then this field is ignored.
+	TargetDuration time.Duration
+
+	// ForceIterations can be used to turn off KDF benchmarking by
+	// setting the time cost directly. If this is zero then the cost
+	// parameters are benchmarked based on the value of TargetDuration.
+	ForceIterations int
+
+	// Parallel sets the maximum number of parallel threads for the
+	// KDF (up to 4). Cryptsetup will adjust this downwards based on
+	// the actual number of CPUs.
+	Parallel int
+}
+
+func (o *KDFOptions) internalOpts() *luks2.KDFOptions {
```

Comment:
given the usage this could return the struct instead of a pointer

Reply:
I've changed this to return a value